File: diagonalDifference/diagonalDifference.js
const arr = [ [ 11, 2, 4 ], [ 4, 5, 6 ], [ 10, 8, -12 ] ]
const arrTwo = [ [ 11, 2, 4, 2 ], [ 4, 5, 6, 1 ], [ 10, 8, -12, 2 ], [5, 7, 3, 1] ]
const arrThree = [[-1, 1, -7, -8], [-10, -8, -5, -2], [0, 9, 7, -1], [4, 4, -2, 1]]

// console.log(arr.flat());
// console.log(arr[2][2]);

function diagonalDifference(arr) {
    let primeDiagonal = 0
    let secondaryDiagonal = 0
    let counter = 0

    for (i=0;i<arr.length;i++) {
        console.log(arr[i][i]);
        primeDiagonal+=arr[i][i]
    }

    for (i=arr.length - 1;i>=0;i--) {
        console.log(arr[i][counter])
        secondaryDiagonal+=arr[i][counter]
        counter+=1

    }

    console.log(Math.abs(primeDiagonal - secondaryDiagonal));
    return Math.abs(primeDiagonal - secondaryDiagonal)


}


// diagonalDifference(arr)
// diagonalDifference(arrTwo)
// diagonalDifference(arrThree)

File: diagonalDifference/diagonal_difference.py
import os
import math
import random
import re
import sys

arr = [ [ 11, 2, 4 ], [ 4, 5, 6 ], [ 10, 8, -12 ] ]
arrTwo = [ [ 11, 2, 4, 2 ], [ 4, 5, 6, 1 ], [ 10, 8, -12, 2 ], [5, 7, 3, 1] ]
arrThree = [[-1, 1, -7, -8], [-10, -8, -5, -2], [0, 9, 7, -1], [4, 4, -2, 1]]

# print(arr[1][1])


def diagonalDifference(arr):
    primary = 0
    secondary = 0
    counter = 0

    for i in range(len(arr)):
        print(arr[i][i])
        primary+=arr[i][i]
    
    for i in reversed(range(len(arr))):
        print(arr[i][counter])
        secondary+=arr[i][counter]
        counter+=1

    print(abs(primary - secondary))
    return abs(primary - secondary)


diagonalDifference(arr)